ban-cooking.lua

```diff
@@ -2,6 +2,33 @@
 -- based on ban-cooking.rb by Putnam: https://github.com/DFHack/scripts/pull/427/files
 -- Putnams work completed by TBSTeun
 
+local options = {}
+
+local argparse = require('argparse')
```

Comment:
These two should stay up here, but the rest of the code you added needs to stay at the bottom. otherwise. `funcs` won't exist before it is called.

In general, you want the order of code to be:

1) requires/reqscripts and global vars (even if they're declared local like `options` is here)
2) classes and functions
3) arg processing and "main" logic. this is last since it needs to call into the classes and functions defined above